### PR TITLE
feat: make region case insensitive

### PR DESCRIPTION
**Pull request by xtreem88**
This PR resolves: https://github.com/customerio/issues/issues/9349

It modifies the logic that checks the region value to be case insensitive
Also, improves the information in the docs

### CODE REVIEW

File: src/ios/withNotificationsXcodeProject.ts [hunk 1]
```diff
@@ -291,9 +291,9 @@ const updateNseEnv = (
 
   if (options.pushNotification?.env?.region) {
     let region = '';
-    if (options.pushNotification?.env?.region === 'us') {
+    if (options.pushNotification?.env?.region?.toLowerCase() === 'us') {
```
Comment (matt-frizzell) [inline src/ios/withNotificationsXcodeProject.ts:294]: Do all our SDKs make this same decision on lower case versus upper case?

**Review by mrehan27 — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -136,6 +136,12 @@ The `customerio-expo-plugin` supports the following configuration options. In mo
             <td><code>false</code></td>
             <td>Enables rich push for iOS</td>
         </tr>
+        <tr>
+            <td style="white-space:nowrap;"><code>ios.pushNotification.env</code></td>
+            <td>object</td>
+            <td><code>undefined</code></td>
+            <td>Set environment variables to use for rich push workaround. This field should be filled when enabling rich push. Expected values: `siteId`: `string`,`apiKey`: `string`, `region`: `us` or `eu`</td>
```
Comment (mrehan27) [inline README.md:143]: Not specific to this PR, but I couldn't find any sample object for this. Should we add an example object so it is clear for customers?

File: src/ios/withNotificationsXcodeProject.ts [hunk 3]
```diff
@@ -291,9 +291,9 @@ const updateNseEnv = (
...
       region = 'Region.US';
-    } else if (options.pushNotification?.env?.region === 'eu') {
+    } else if (options.pushNotification?.env?.region?.toLowerCase() === 'eu') {
       region = 'Region.EU';
     }
```
Comment (Shahroz16) [inline src/ios/withNotificationsXcodeProject.ts:298]: Can we maybe avoid the changing? how does this look? 

```suggestion
     const regionMap = {
  us: 'Region.US',
  eu: 'Region.EU',
};

const region = options.pushNotification?.env?.region?.toLowerCase();
const mappedRegion = regionMap[region] || '';
```

File: README.md (hunk 2 above)
Comment (xtreem88) [reply README.md:143]: This is part of the docs cleanup we will be doing soon with @glosier's help

**Review by Shahroz16 — APPROVED** [review]

**Pull request merged**